src/components/FeatureStyler/FeatureStyler.js

```diff
@@ -497,6 +591,30 @@ class FeatureStyler extends PureComponent {
       textRotation = featStyle.getText().getRotation();
     }
 
+    const oldStyles = FeatureStyler.getStyleAsArray(feature);
+
+    if (oldStyles.length > 1) {
+      for (let i = 1; i < oldStyles.length; i += 1) {
+        if (oldStyles[i].getImage() && oldStyles[i].getImage().getSrc()) {
+          for (let j = 0; j < lineIcons.length; j += 1) {
+            if (oldStyles[i].getImage().getSrc() === lineIcons[j].icon) {
+              const coord = oldStyles[i].getGeometry().getCoordinates();
+              const startCoord = FeatureStyler.getVertexCoord(
+                feature.getGeometry(),
+              );
+              if (coord[0] === startCoord[0] && coord[1] === startCoord[1]) {
+                if (!lineStartIcon) {
+                  this.setState({ lineStartIcon: lineIcons[j].icon });
+                }
+              } else if (!lineEndIcon) {
+                this.setState({ lineEndIcon: lineIcons[j].icon });
```

Comment:
`setState` is already called at the end, can you use this call with the lineStaertIcon lineEndIcon values

Reply:
done